Multisig/UI/WalletConnectV2/WalletConnectManager.swift

```diff
@@ -183,15 +199,98 @@ class WalletConnectManager {
         precondition(Thread.isMainThread)
         Safe.removeSession(topic: topic)
     }
-    
+
     func getSessions(topics: [String]) -> [Session] {
-        Sign.instance.getSessions().filter({topics.contains($0.topic)})
+        Sign.instance.getSessions().filter({ topics.contains($0.topic) })
     }
+
+    private func handle(request: Request) {
+        dispatchPrecondition(condition: .notOnQueue(.main))
+
+        guard let session = getSessions(topics: [request.topic]).first else {
+            reject(request: request)
+            return
+        }
+
+
+        guard let safe = Safe.by(topic: request.topic) else {
```

Comment:
If you "Safe.by(topic)" to be performed on main thread then this one should be performed on main thread, that's why I made the original implementation as it was.